### PR TITLE
Release Godot 4.2 beta 3 and publish release notes

### PR DESCRIPTION
**Pull request by akien-mga**

### CODE REVIEW

**Comment by raulsntos** [comment]
Would it be OK to encourage users to use the .NET 8 preview so we can get some testing before the .NET stable release this November? I was thinking about adding a note in the intro mentioning that .NET 8.0 should be supported and encouraging users to test it and report the bugs they find.

**Comment by akien-mga** [comment]
Yeah that's a good idea. Do you want to suggest changes to the intro and to the download section to match this?

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Since the start of the beta phase, we fixed a number of bugs reported by the community, so we're publishing further beta snapshots to validate those fixes, and iterate closer to the release candidate stage.
+
+This third beta notably fixes a porting issue for 3D Godot 4.1 projects following our changes to the [mesh vertices and attributes compression format](https://godotengine.org/article/dev-snapshot-godot-4-2-beta-1/#rendering-particles-and-shaders). This beta provides a helpful dialog when you first open a pre-existing project, that lets you choose to upgrade and re-save all the meshes in your projects to make them compatible with Godot 4.2.
+
+This release also [updates the toolchains](https://github.com/godotengine/build-containers/pull/128) used to build official binaries, notably for Windows, macOS, iOS and JavaScript. This should be mostly transparent to end users, but toolchain bugs are a possibility, so please report if anything seems off compared to the previous beta build in terms of performance, or outright crashing.
+
```
Comment (raulsntos) [inline collections/_article/dev-snapshot-godot-4-2-beta-3.md:21]: Maybe something like this, idk.

```suggestion

For C# users, the .NET 8.0 release is set to be launched this November. Make sure to try the current preview release ([`8.0.0-rc.2`](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)) with Godot to make sure everything works as expected. This Godot beta release includes a fix to [support preview releases](https://github.com/godotengine/godot/pull/83325) so give it a try and report any bugs that you find.

```

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+## Downloads
+
+The downloads for this pre-release build can be found in our GitHub repository:
+
+* [**Download Godot 4.2 beta 3**](https://github.com/godotengine/godot-builds/releases/tag/4.2-beta3).
+
+**Standard build** includes support for GDScript and GDExtension.
+
+**.NET build** (marked as `mono`) includes support for C#, as well as GDScript and GDExtension.
+- .NET build requires [.NET SDK 6.0](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) or [7.0](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) installed in a standard location.
```
Comment (raulsntos) [inline collections/_article/dev-snapshot-godot-4-2-beta-3.md:106]:
```suggestion
- .NET build requires [.NET SDK 6.0](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) or [7.0](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) installed in a standard location.
  - [.NET 8.0](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) should also be supported, but keep in mind this version of the .NET SDK is still in preview. Give it a try and let us know if you find any bugs.
```

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
+---
+title: "Dev snapshot: Godot 4.2 beta 3"
+excerpt: "We continue iterating quickly on beta snapshots for Godot 4.2, to ensure that we can solve regressions before the stable release."
```
Comment (clayjohn) [inline collections/_article/dev-snapshot-godot-4-2-beta-3.md]:
```suggestion
excerpt: "We continue iterating quickly on beta snapshots for Godot 4.2 to ensure that we can solve regressions before the stable release."
```

**Review by clayjohn — COMMENTED** [review]
Just one grammar nitpick

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+image_caption_title: The Last Game
+image_caption_description: A game by Frédéric Julian
+date: 2023-10-24 15:00:00
+---
+
+We entered the beta phase for Godot 4.2 a couple of weeks ago, with [4.2 beta 1](/article/dev-snapshot-godot-4-2-beta-1/). If you missed that release, have a look at the [release notes](/article/dev-snapshot-godot-4-2-beta-1/) for an overview of the key changes in Godot 4.2.
+
+Since the start of the beta phase, we fixed a number of bugs reported by the community, so we're publishing further beta snapshots to validate those fixes, and iterate closer to the release candidate stage.
+
+This third beta notably fixes a porting issue for 3D Godot 4.1 projects following our changes to the [mesh vertices and attributes compression format](https://godotengine.org/article/dev-snapshot-godot-4-2-beta-1/#rendering-particles-and-shaders). This beta provides a helpful dialog when you first open a pre-existing project, that lets you choose to upgrade and re-save all the meshes in your projects to make them compatible with Godot 4.2.
```
Comment (YuriSizov) [inline collections/_article/dev-snapshot-godot-4-2-beta-3.md]: ~Hmm, what makes it a porting issue?~ Okay, I got it now, but at first I read "porting" in the platform support sense. Maybe there is a way to phrase it better.

Also two sentences in the row starting with "This beta".

```suggestion
Notably, this third beta improves support for 3D projects made with Godot 4.1 when migrating them to Godot 4.2. Following our changes to the [mesh vertices and attributes compression format](https://godotengine.org/article/dev-snapshot-godot-4-2-beta-1/#rendering-particles-and-shaders), we are now introducing a helpful dialog when you first open a pre-existing project. It lets you choose to upgrade and re-save all the meshes in your projects to make them compatible with Godot 4.2.
```

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+This release also [updates the toolchains](https://github.com/godotengine/build-containers/pull/128) used to build official binaries, notably for Windows, macOS, iOS and JavaScript. This should be mostly transparent to end users, but toolchain bugs are a possibility, so please report if anything seems off compared to the previous beta build in terms of performance, or outright crashing.
```
Comment (YuriSizov) [inline collections/_article/dev-snapshot-godot-4-2-beta-3.md]:
```suggestion
This release also [updates the toolchains](https://github.com/godotengine/build-containers/pull/128) used to build official binaries, notably for Windows, macOS, iOS and JavaScript. This should be mostly transparent to end users, but toolchain bugs are a possibility. So please report if anything seems off compared to the previous beta build in terms of performance, or outright crashing.
```

I would try for shorter sentences :)

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+[Jump to the **Downloads** section](#downloads), and give it a spin right now, or continue reading to learn more about improvements in this release. You can also [try the **Web editor**](https://editor.godotengine.org/releases/4.2.beta3/) or the **Android editor** for this release. If you are interested in the latter, please request to join [our testing group](https://groups.google.com/g/godot-testers) to get access to pre-release builds.
+
+-----
+
+*The illustration picture for this article is taken from* [**The Last Game**](https://twitter.com/TheLastGame__), *a roguelite twin-stick shooter by Frédéric Julian made with Godot 4.1. It was recently released on [App Store](https://apps.apple.com/us/app/the-last-game/id6466296603), [Google Play](https://play.google.com/store/apps/details?id=com.frju.thelastgame), [Steam](https://store.steampowered.com/app/2563800/The_Last_Game/) and [itch.io](https://frju.itch.io/the-last-game). You can follow the development on [Twitter](https://twitter.com/TheLastGame__) or [Discord](https://discord.gg/J4VQtvEb9W).*
```
Comment (YuriSizov) [inline collections/_article/dev-snapshot-godot-4-2-beta-3.md]:
```suggestion
*The illustration picture for this article is taken from* [**The Last Game**](https://twitter.com/TheLastGame__), *a roguelite twin-stick shooter by Frédéric Julian made with Godot 4.1. It was recently released on [App Store](https://apps.apple.com/us/app/the-last-game/id6466296603), [Google Play](https://play.google.com/store/apps/details?id=com.frju.thelastgame), [Steam](https://store.steampowered.com/app/2563800/The_Last_Game/), and [itch.io](https://frju.itch.io/the-last-game). You can follow the development on [Twitter](https://twitter.com/TheLastGame__) or [Discord](https://discord.gg/J4VQtvEb9W).*
```

Oxford comma?

**Review by YuriSizov — APPROVED** [review]
Some small adjustments, but should be good to go once other concerns are addressed.

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md (hunk 3 above)
Comment (YuriSizov) [reply collections/_article/dev-snapshot-godot-4-2-beta-3.md]: I think this was to emphasize the second part, maybe a dash is more appropriate?

**Review by mhilbrunner — APPROVED** [review]
LGTM besides what was noted above

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md (hunk 3 above)
Comment (akien-mga) [reply collections/_article/dev-snapshot-godot-4-2-beta-3.md]: I think both are fine, so I'm good removing the comma.

File: collections/_article/dev-snapshot-godot-4-2-beta-3.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+- Rendering: Some more fixes for compressed meshes ([GH-83704](https://github.com/godotengine/godot/pull/83704)).
+- Rendering: Fix `GPUParticles3D` on the Meta Quest 2 with OpenGL renderer ([GH-83756](https://github.com/godotengine/godot/pull/83756)).
+- Rendering: Fix LightmapGI taking editor-only and sky-only lights into account ([GH-83861](https://github.com/godotengine/godot/pull/83861)).
+- Thirdparty: brotli: Sync with upstream 1.1.0 ([GH-82580](https://github.com/godotengine/godot/pull/82580)).
+- Thirdparty: ThorVG: update to v0.11.2 ([GH-83656](https://github.com/godotengine/godot/pull/83656)).
+- Thirdparty: mbedTLS: Update to version 2.18.5 ([GH-83721](https://github.com/godotengine/godot/pull/83721)).
+- Thirdparty: openxr: Revert to 1.0.28, newer versions crash on Windows/Mingw-GCC ([GH-83829](https://github.com/godotengine/godot/pull/83829)).
+
+This release is built from commit [`e8d57afae`](https://github.com/godotengine/godot/commit/e8d57afaeccf0d9f9726746f49936eb93aa0039b) (see [README](https://github.com/godotengine/godot-builds/releases/download/4.2-beta3/README.txt)).
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-2-beta-3.md]:
```suggestion
This release is built from commit [`e8d57afae`](https://github.com/godotengine/godot/commit/e8d57afaeccf0d9f9726746f49936eb93aa0039b).
```

**Pull request merged**